src/v/iceberg/avro_decimal.h

```diff
@@ -27,9 +27,9 @@ inline bytes encode_avro_decimal(absl::int128 decimal) {
 
     for (int i = 0; i < 8; i++) {
         // NOLINTNEXTLINE(hicpp-signed-bitwise)
-        decimal_bytes[i] = (low_half >> (i * 8)) & 0xFF;
+        decimal_bytes[i] = (high_half >> ((7 - i) * 8)) & 0xFF;
         // NOLINTNEXTLINE(hicpp-signed-bitwise)
-        decimal_bytes[i + 8] = (high_half >> (i * 8)) & 0xFF;
+        decimal_bytes[i + 8] = (low_half >> ((7 - i) * 8)) & 0xFF;
```

Comment:
This assumes LE on our end. can we call `ss::cpu_to_be` and then write that out?